Server/src/main/java/org/xdi/oxauth/authorize/ws/rs/AuthorizeRestWebServiceImpl.java

```diff
@@ -187,7 +187,23 @@ public Response requestAuthorization(
         try {
             Map<String, String> customResponseHeaders = Util.jsonObjectArrayStringAsMap(customRespHeaders);
 
-            sessionIdService.assertAuthenticatedSessionCorrespondsToNewRequest(sessionUser, acrValuesStr);
+            try {
+                sessionIdService.assertAuthenticatedSessionCorrespondsToNewRequest(sessionUser, acrValuesStr);
+            } catch (AcrChangedException e) { // Acr changed
+                //See https://github.com/GluuFederation/oxTrust/issues/797
+                if (e.isMethodEnabled()) {
+                    if (!prompts.contains(Prompt.LOGIN)) {
+                        log.info("ACR is changed, adding prompt=login to prompts");
+                        prompts.add(Prompt.LOGIN);
+                        String sessionPrompts = sessionUser.getSessionAttributes().get("prompt");
+                        sessionPrompts = StringUtils.isEmpty(sessionPrompts) ? "" : (sessionPrompts + " ");
+                        //Override prompt in session
+                        sessionUser.getSessionAttributes().put("prompt", sessionPrompts + Prompt.LOGIN.getParamName());
+                    }
```

Comment:
it seems we can put it just with one line if I don't miss anything `org.xdi.oxauth.model.util.StringUtils.implode`
```
 sessionUser.getSessionAttributes().put("prompt", StringUtils.implode(prompts, " ");
```

Reply:
yes, as long as `prompt` session attribute contains the same info of `prompts` List (in other words the `prompt` parameter of `requestAuthorization` method). Are those equivalent?

Reply:
Actually on the code above you added `login`, means it can't be equivalent.

Reply:
obviating that fact, of course... are they equivalent?

Reply:
It may or may not be equivalent. Our goal here is to force user re-authenticate. It means we set our session as NOT authenticated and added `LOGIN` to prompt. `prompts` array is constructed from request `prompt` parameter. Since we are re-authenticating the user, new value must be taken from request (with LOGIN value).